### PR TITLE
Autosizing modes

### PR DESCRIPTION
**Pull request by kylebgorman**
In this follow-up to #194, I make it so that the batch size finder has two modes:

* `opt`imal mode is the same as before: it uses the gradient accumulation trick if necessary
* `max` mode just finds the max batch size and uses it, ignoring `--batch_size`

I theorize the latter would be really useful for when you have reason to believe the model does well with large batches (this might be all models?) and rather than tuning it, want to tune the other hyperparameters instead.

I am testing this with a fancy Bayesian sweep [here](https://wandb.ai/hyperparam-sensitivity/pol-max_batch-hard_monotonic_lstm?nw=nwuserkylebgorman) (the link may not be accessible to just anyone).

### CODE REVIEW

**Review by Adamits — APPROVED** [review]
LGTM
